repository/inventory_repo.go

```diff
@@ -134,3 +180,108 @@ func (i InventoryRepo) GetKeysForEnvironment(ctx context.Context, env string) (m
 	}
 	return scan, nil
 }
+
+func (i InventoryRepo) BuildNotifications(assets domain.Assets) []domain.SSEMessage {
+	var events []domain.SSEMessage
+	events = append(events, getDeleteEvents(assets.Deleted)...)
+	events = append(events, getCreateEvents(assets.Created)...)
+	// TODO: this currently sends patch notification for all flags
+	// regardless weather they have changed or not
+	events = append(events, getPatchEvents(assets.Patched)...)
+	return events
+}
+
+func getDeleteEvents(m map[string]string) []domain.SSEMessage {
+	res := make([]domain.SSEMessage, 0, len(m))
+	if m == nil {
+		return []domain.SSEMessage{}
+	}
+	for k := range m {
+		if strings.Contains(k, featureVariant) {
+			res = append(res, parseFlagEntry(k, deleteVariant))
+		}
+		if strings.Contains(k, segmentVariant) {
+			res = append(res, parseSegmentEntry(k, deleteVariant))
+		}
+	}
+	return res
+}
+
+func getCreateEvents(m map[string]string) []domain.SSEMessage {
+	res := make([]domain.SSEMessage, 0, len(m))
+	if m == nil {
+		return []domain.SSEMessage{}
+	}
+	for k := range m {
+		if strings.Contains(k, featureVariant) {
+			res = append(res, parseFlagEntry(k, createVariant))
+		}
+		if strings.Contains(k, segmentVariant) {
+			res = append(res, parseSegmentEntry(k, createVariant))
+		}
+	}
+	return res
+}
+
+func getPatchEvents(m map[string]string) []domain.SSEMessage {
+	res := make([]domain.SSEMessage, 0, len(m))
+	if m == nil {
+		return []domain.SSEMessage{}
+	}
+	for k := range m {
+		if strings.Contains(k, featureVariant) {
+			res = append(res, parseFlagEntry(k, patchVariant))
+		}
+		if strings.Contains(k, segmentVariant) {
+			res = append(res, parseSegmentEntry(k, patchVariant))
+		}
+	}
+	return res
+}
+
+func parseFlagEntry(flagString, variant string) domain.SSEMessage {
+	env, id, err := parseFlagString(flagString)
+	if err != nil {
+		log.Error(err)
+		return domain.SSEMessage{}
+	}
+	return domain.SSEMessage{
+		Domain:      "flag",
+		Event:       variant,
+		Identifier:  id,
+		Environment: env,
+		Version:     0,
+	}
+}
+func parseSegmentEntry(segmentString, variant string) domain.SSEMessage {
+	env, id, err := parseSegmentString(segmentString)
+	if err != nil {
+		log.Error(err)
+		return domain.SSEMessage{}
+	}
+	return domain.SSEMessage{
+		Domain:      "target-segment",
+		Event:       variant,
+		Identifier:  id,
+		Environment: env,
+		Version:     0,
+	}
+}
+
+func parseFlagString(flagString string) (string, string, error) {
+	re := regexp.MustCompile(`env-([a-zA-Z0-9-]+)-feature-config-([a-zA-Z0-9-]+)`)
```

Comment:
This is a minor thing but if we declare these as variables at the package level then we only have to compile the regex once at start up. But if we leave them in here we have to compile it every time this function is called. Only pointing it out because we did something like this in one of our services, can't remember which but it showed up as being quite expensive and moving it to the package level reduced CPU usage a good bit.